sdk-exporter/trace/shared/src/main/scala/org/typelevel/otel4s/sdk/exporter/otlp/OtlpHttpSpanExporter.scala

```diff
@@ -0,0 +1,221 @@
+import org.http4s.ember.client.EmberClientBuilder
+import org.http4s.syntax.literals._
+import org.typelevel.ci._
+import org.typelevel.otel4s.sdk.exporter.otlp.OtlpHttpSpanExporter.Encoding
+import org.typelevel.otel4s.sdk.trace.data.SpanData
+import org.typelevel.otel4s.sdk.trace.exporter.SpanExporter
+
+import scala.concurrent.duration._
+
+private final class OtlpHttpSpanExporter[F[_]: Temporal: Console] private (
+    client: Client[F],
+    config: OtlpHttpSpanExporter.Config
+) extends SpanExporter[F] {
+  import JsonCodecs._
+
+  val name: String = {
+    val headers = config.headers.mkString(
+      "headers={",
+      ",",
+      "}",
+      Headers.SensitiveHeaders
+    )
+
+    "OtlpHttpSpanExporter{" +
+      s"encoding=${config.encoding}, " +
+      s"endpoint=${config.endpoint}, " +
+      s"timeout=${config.timeout}, " +
+      s"gzipCompression=${config.gzipCompression}, " +
+      headers +
+      "}"
+  }
+
+  private implicit val spansEncoder: EntityEncoder[F, List[SpanData]] =
+    config.encoding match {
+      case Encoding.Json =>
+        import org.http4s.circe._
+        jsonEncoderOf[F, List[SpanData]]
+
+      case Encoding.Protobuf =>
+        val content = Header.Raw(ci"Content-Type", "application/x-protobuf")
+        EntityEncoder.simple(content) { spans =>
+          val request: ExportTraceServiceRequest = ProtoCodecs.toProto(spans)
+          Chunk.array(request.toByteArray)
+        }
+    }
+
+  def exportSpans[G[_]: Foldable](spans: G[SpanData]): F[Unit] = {
+    val request =
+      Request[F](Method.POST, config.endpoint, HttpVersion.`HTTP/1.1`)
+        .withEntity(spans.toList)
+        .putHeaders(config.headers)
+
+    client
+      .run(request)
+      .use(response => logBody(response).unlessA(response.status.isSuccess))
+      .timeout(config.timeout)
+      .handleErrorWith { e =>
+        Console[F].errorln(
+          s"OtlpHttpSpanExporter: cannot export spans: ${e.getMessage}\n${e.getStackTrace.mkString("\n")}\n"
+        )
+      }
+  }
+
+  def flush: F[Unit] = Temporal[F].unit
+
+  private def logBody(response: Response[F]): F[Unit] =
+    for {
+      body <- response.bodyText.compile.string
+      _ <- Console[F].println(
+        s"[OtlpHttpSpanExporter/${config.encoding}] the request failed with [${response.status}]. Body: $body"
+      )
+    } yield ()
+
+}
+
+object OtlpHttpSpanExporter {
+
+  private object Defaults {
+    val Endpoint: Uri = uri"http://localhost:4318/v1/traces"
+    val Timeout: FiniteDuration = 10.seconds
+    val GzipCompression: Boolean = false
+  }
+
+  private final case class Config(
+      encoding: Encoding,
+      endpoint: Uri,
+      timeout: FiniteDuration,
+      headers: Headers,
+      gzipCompression: Boolean
+  )
+
+  sealed trait Encoding
+  object Encoding {
+    case object Json extends Encoding
+    case object Protobuf extends Encoding
+  }
+
+  /** A builder of [[OtlpHttpSpanExporter]] */
+  sealed trait Builder[F[_]] {
```

Comment:
Some settings are missing: `withRetryPolicy`, `withTrustedCertificates`, `withClientTls`, `withSslContext`.
Currently, I plan to have a minimal proof-of-concept client that works.